scipy/special/_cosine_distr.pxd

```diff
@@ -0,0 +1,244 @@
+# -*- coding: utf-8 -*-
+
+"""
+The functions
+
+    cosine_cdf
+    cosine_invcdf
+
+defined here are the kernels for the ufuncs
+
+    _cosine_cdf
+    _cosine_invcdf
+
+defined in scipy.special._funcs.
+
+The ufuncs are used by the class scipy.stats.cosine_gen.
+
+"""
+
+from libc.math cimport sin, cos, fabs, M_PI, NAN
+
+
+# p and q (below) are the coefficients in the numerator and denominator
+# polynomials (resp.) of the Pade approximation of
+#     f(x) = (pi + x + sin(x))/(2*pi)
+# at x=-pi.  The coefficients are ordered from lowest degree to highest.
+# These values are used in the function cosine_cdf_pade_approx_at_neg_pi(x).
+#
+# These coefficients can be derived by using mpmath as follows:
```

Comment:
Instead of documenting here, you could add an entry to `special._precompute` that outputs the desired coefficients.

Reply:
I added a short script with this code to the `_precomute` directory.